src/github/graphql.ts

```diff
@@ -15,11 +16,14 @@ export interface GraphQLError {
 }
 
 export function addGraphQL (client: GitHubAPI) {
-  const graphql = octokitGraphqlWithDefaults(client.request, {
-    method: 'POST',
-    url: '/graphql',
+  // TODO: We should not be asserting types. The `defaults` method exists on
+  // `client.request`, but is not reflected by the definitions for
+  // `Octokit.Request`
+  const graphqlRequest = (client.request as any as typeof request).defaults({
```

Comment:
what does as `as any as typeof request` mean? 

And I think `client.request` should already have the right `baseUrl` set, we should not need to set it again.

Reply:
The `as any as typeof request` is asserting the actual type of `client.request`. TypeScript believes `client.request` is of type `Octokit.Request` (from `@Octokit/rest`). `client.request` is actually `typeof request`, from [@Octokit/request](https://github.com/octokit/graphql.js/blob/d457a93b87cdc233bce1e8a0b7149b850fdb660e/package.json#L31).

We're asserting the correct type `client.request` so we have accurate typing information to access the `defaults` property.


With regards to `baseUrl`, `graphql.test.js` would fail without it.

Reply:
> With regards to `baseUrl`, `graphql.test.js` would fail without it.

I need to look into why that is. Might be a problem with how the tests are setup, I'll look into it

Reply:
Okay I see it now. This code is such a mess lol, cannot wait to clean this up with all the new Octokit libraries.

Thanks Maxim, I'll get this PR merged